lambdas/cfn-sc-actions-provider/README.md

```diff
@@ -0,0 +1,129 @@
+# Service Catalog Actions Provider CloudFormation Template Custom Resource
+
+This template and associated Lambda function and custom resource add support to AWS CloudFormation for the [AWS Service Catalog Service Actions][1]
+resource type.
+
+# Lambda
+We create the lambda in AWS to provide SC action functionality
+
+## Build Lambda
+Build the lambda using the [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+
+```shell script
+sam build --build-dir lambdas/build/cfn-sc-actions-provider --base-dir lambdas/cfn-sc-actions-provider --template lambdas/cfn-sc-actions-provider/template.yaml
+```
+
+## Deploy Lambda to S3
+
+```shell script
+sam package --template-file lambdas/build/cfn-sc-actions-provider/template.yaml --s3-bucket essentials-awss3lambdaartifactsbucket-x29ftznj6pqw --output-template-file templates/cfn-sc-actions-provider.yaml
+```
+
+## Install Lambda into AWS
+Create the following [sceptre](https://github.com/Sceptre/sceptre) file
+
+config/prod/cfn-sc-actions-provider.yaml
+```yaml
+template_path: "remote/cfn-sc-actions-provider.yaml"
+stack_name: "cfn-sc-actions-provider"
+hooks:
+  before_launch:
+    - !cmd "curl https://s3.amazonaws.com/{{stack_group_config.admincentral_cf_bucket}}/aws-infra/master/cfn-sc-actions-provider.yaml --create-dirs -o templates/remote/cfn-sc-actions-provider.yaml"
+```
+
+Install the lambda using sceptre:
+```bash script
+sceptre --var "profile=my-profile" --var "region=us-east-1" launch prod/cfn-sc-actions-provider.yaml
+```
+
+# Service Catalog Action
+We create the SC actions by using the cfn-sc-actions-provider lambda.  The provider contains two custom
+resources, one to create the SC action and one to associate the action to a product.
+
+## Creating SC Actions and Asssociation
+Create the sceptre file
+
+config/prod/sc-restart-instance-action.yaml:
+```yaml
+template_path: sc-action.yaml
+stack_name: sc-restart-instance-action
+stack_tags:
+  Department: "Platform"
+  Project: "Infrastructure"
+  OwnerEmail: "joe.smith@sagebase.org"
+parameters:
+  Name: "RestartEC2Instance"
+  SsmDocName: "AWS-RestartEC2Instance"
+  SsmDocVersion: "1"
```

Comment:
what is the purpose of this version? When will it change?

Reply:
SSM doc can have multiple versions, this is the version number.  check the `events/actions.txt` file for more info and here's the docs on ssm automation docs feature: https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-documents-reference-details.html